components/datetime/src/provider/neo.rs

```diff
@@ -357,6 +365,8 @@ impl SimpleSubstitutionPattern<'_> {
     }
 }
 
+check_size!(LinearNamesV1, linear_names_v1_size, 32 | 24);
```

Comment:
thought (non blocking) : I know we have a lot of these `|` tests but I would kind of like to have them be smarter about Rust versions so we know if things slide back (or whatever).

This is pretty hard to do though, so probably not worth it? We should potentially try and whittle them down every time we update MSRV.


One way to achieve this would be for `check_size!()` to _only_ check against the last variant if `cfg(__pinned_rust)` is set, and set that CFG in our `rust-toolchain` tests.

Reply:
I like the `cfg` idea! Check it out in the latest push.

Reply:
No, we shouldn't be disabling the test _entirely_: if a nightly changes something we should definitely nottice..

My proposal was that we allow for the `|` syntax, but if someone uses it, on `icu4x_check_sizes` mode, we only check against the last (or first?) one. Or we have a separate syntax like `default = 32, all = 12 | 18 | 32`

Reply:
The problem I have with the `|` tests is that the assertion could lag behind and it doesn't really test anything; there's no way to know in review which part of the `|` is the correct one.

There are only 3 toolchain versions worth testing:

1. MSRV
2. Pinned Stable
3. Latest Nightly

Currently I'm testing pinned stable. I can see a use case behind testing latest nightly. Testing MSRV is only useful for documentation purposes.

How about if I change the macro so that you can either write a single number `32` or the syntax `stable = 32, nightly = 24`, and then the test uses the nightly version number if `ICU4X_FORCED_NIGHTLY_TOOLCHAIN` is set?

I still have a slight preference for setting the test as `#[ignore]` by default because I just don't really want people coming to us saying that we have a failing test. Brittle tests are best to disable except when the environment is specially designed to make them pass.

Reply:
Yeah, I'm in favor of `icu4x_pinned = 32, nightly = 24` and then turning it off by default for other modes. That seems ideal. And I agree: tests like this are best not exposed to third parties attempting to run our code.

We should make sure the assert tells people to build with `cfg(whatever)` when testing locally otherwise folks will be unable to repro failing tests.

Reply:
strawman syntax: `size_test!(...., 24)` is "both", otherwise you need to use `size_test!(...., pinned_stable = .., nightly = ..)`. The panic talks about all of this.